third_party/gfxreconstruct/framework/decode/vulkan_dive_consumer_base.h

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2025 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef GFXRECON_DECODE_VULKAN_DIVE_CONSUMER_BASE_H
+#define GFXRECON_DECODE_VULKAN_DIVE_CONSUMER_BASE_H
+
+#include "util/output_stream.h"
+#include "util/defines.h"
+#include "annotation_handler.h"
+#include "format/platform_types.h"
+#include "generated/generated_vulkan_consumer.h"
+#include "vulkan/vulkan.h"
+#include "util/json_util.h"
+#include "util/dive_function_data.h"
+
+#include <cstdint>
+#include <cstdio>
+#include <string>
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(decode)
+
+class VulkanExportDiveConsumerBase : public VulkanConsumer
+{
+  public:
+    VulkanExportDiveConsumerBase() = default;
+
+    virtual ~VulkanExportDiveConsumerBase() override;
+
+    void Initialize(AnnotationHandler* writer);
+
+    void Destroy();
+
+    bool IsValid() const { return true; }
```

Comment:
is this func needed if it is always returning true?

Reply:
Removed.